drivers/sensor/hts221/hts221_trigger.c

```diff
@@ -121,33 +118,46 @@ int hts221_init_interrupt(const struct device *dev)
 {
 	struct hts221_data *data = dev->data;
 	const struct hts221_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	int status;
+
+	/* setup data ready gpio interrupt */
+	if (!device_is_ready(cfg->gpio_drdy.port)) {
+		/* API may return false even when ptr is NULL */
+		if (cfg->gpio_drdy.port != NULL) {
+			LOG_ERR("device %s is not ready", cfg->gpio_drdy.port->name);
+			return -ENODEV;
+		}
+
+		LOG_DBG("gpio_drdy not defined in DT");
+		return 0;
+	}
```

Comment:
may read better if:

```c 
	if (cfg->gpio_drdy.port == NULL) {
		LOG_DBG("gpio_drdy not defined in DT");
		return 0;
	}

	if (!device_is_ready(cfg->gpio_drdy.port)) {
		LOG_ERR("device %s is not ready", cfg->gpio_drdy.port->name);
		return -ENODEV;
	}
```

Reply:
I just c/p what has been done in other drivers but indeed it's more logical to read it that way